include/powsybl/math/UndirectedGraph.hpp

```diff
@@ -54,19 +54,29 @@ class UndirectedGraph {
 
     const stdcxx::Reference<E>& getEdgeObject(unsigned long e) const;
 
+    const_range<E> getEdgeObjectsConnectedToVertex(unsigned long vertex) const;
+
+    const std::vector<unsigned long>& getEdgeConnectedToVertex(unsigned long vertex) const;
+
+    const std::vector<unsigned long>& getEdgesConnectedToVertex(unsigned long vertex) const;
+
     const_range<E> getEdgeObjects() const;
 
     range<E> getEdgeObjects();
 
     const_range<E> getEdgeObjects(unsigned long v1, unsigned long v2) const;
 
+    const unsigned long& getEdgeVertex1(unsigned long edge) const;
+
+    const unsigned long& getEdgeVertex2(unsigned long edge) const;
+
     const_range<unsigned long> getEdges() const;
 
     unsigned long getMaxVertex() const;
 
-    unsigned long getVertex1(unsigned long e) const;
+    const unsigned long& getVertex1(unsigned long e) const;
```

Comment:
To be discussed: I think it's not wrong, I had to update that in order to return a range in `NodeBreakerViewImpl::getNodesInternalConnectedTo` (unless it was gardening...). But maybe `NodeBreakerViewImpl::getNodesInternalConnectedTo` should return a vector<unsigned long> instead ?